### PR TITLE
Core/Player: cleanup function CheckInstanceLoginValid

### PR DESCRIPTION
**Pull request by ForesterDev**
CleanUP and optimize checks added in https://github.com/TrinityCore/TrinityCore/commit/5f917e2286809d7aafa8871be38042ae7966c26b

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -18672,31 +18672,31 @@ bool Player::Satisfy(AccessRequirement const* ar, uint32 target_map, bool report
...
 
     // do checks for satisfy accessreqs, instance full, encounter in progress (raid), perm bind group != perm bind player
-    return sMapMgr->CanPlayerEnter(map->GetId(), this, true) || IsInstanceLoginGameMasterException();
+    return sMapMgr->CanPlayerEnter(map->GetId(), this, true);
 }
 
 bool Player::IsInstanceLoginGameMasterException() const
 {
-    if (CanBeGameMaster())
+    if (IsGameMaster() || CanBeGameMaster())
```
Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp]: With CanBeGameMaster you don't need to check IsGameMaster - CanBeGameMaster() checks if you have the rbac permission to use .gm on, while IsGameMaster() returns true if you used .gm on

File: src/server/game/Entities/Player/Player.cpp [hunk 2]
```diff
@@ -18672,31 +18672,31 @@ bool Player::Satisfy(AccessRequirement const* ar, uint32 target_map, bool report
 
 bool Player::CheckInstanceLoginValid(Map* map)
 {
-    if (!map->IsDungeon() || IsGameMaster())
+    if (!map->IsDungeon() || IsInstanceLoginGameMasterException())
```
Comment (jackpoz) [inline src/server/game/Entities/Player/Player.cpp]: Isn't this going to send the ingame message even if the player is actually in a raid/group ?

Please provide a list of the tests you did ingame.

Comment (ForesterDev) [inline src/server/game/Entities/Player/Player.cpp]: Yes, player will receive message even if he is in raid group. But we can send this warning-message only if player is not in group, if needed, just add (!GetGroup()) check in IsInstanceLoginGameMasterException

File: src/server/game/Entities/Player/Player.cpp [hunk 3]
```diff
@@ -2563,7 +2563,7 @@ void Player::SetGameMaster(bool on)
 
 bool Player::CanBeGameMaster() const
 {
-    return m_session && m_session->HasPermission(rbac::RBAC_PERM_COMMAND_GM);
+    return GetSession()->HasPermission(rbac::RBAC_PERM_COMMAND_GM);
```
Comment (ccrs) [inline src/server/game/Entities/Player/Player.cpp]: is this safe?

Comment (Shauren) [inline src/server/game/Entities/Player/Player.cpp]: Yes, player session will never be null.

File: src/server/game/Entities/Player/Player.cpp (hunk 2 above)
Comment (jackpoz) [inline src/server/game/Entities/Player/Player.cpp]: That will duplicate the group checks, the idea behind IsInstanceLoginGameMasterException() is to replace any previous "return false;" in Player::CheckInstanceLoginValid() .
I'd like to merge this PR without the changes in Player::CheckInstanceLoginValid(), the rest of the PR is ok

File: sql/updates/world/2015_11_27_00_world_355.sql [hunk 4]
```diff
@@ -0,0 +1,3 @@
+DELETE FROM `trinity_string` WHERE `entry` = 11010;
```
Comment (jackpoz) [inline sql/updates/world/2015_11_27_00_world_355.sql]: this file should be called 335 btw, not 355

**Pull request merged**